common/src/lib.rs

```diff
@@ -318,34 +331,48 @@ pub fn get_program(id: H256) -> Option<Program> {
 }
 
 /// Returns mem page data from storage for program `id` and `page_idx`
-pub fn get_program_page_data(
-    id: H256,
-    page_idx: PageNumber,
-) -> Option<Result<PageBuf, MemoryError>> {
-    let key = page_key(id, page_idx);
-    let data = sp_io::storage::get(&key)?;
-    Some(PageBuf::new_from_vec(data))
+pub fn get_program_page_data(program_id: H256, page: PageNumber) -> Result<PageBuf, CommonError> {
+    let key = page_key(program_id, page);
+    let data =
+        sp_io::storage::get(&key).ok_or(CommonError::CannotFindDataForPage { program_id, page })?;
+    PageBuf::new_from_vec(data).map_err(Into::into)
 }
 
+/// Returns data for all `program` pages, which has data
```

Comment:
```suggestion
/// Returns data for all `program` pages, which contain data.
```

Reply:
In this case `has` is more suitable. Because storage `contains` data and page has data in storage

Reply:
in English after `which` we must use `has`

Reply:
`which have` is correct form for plural nouns, but not such popular as `that have`
[src](https://www.quora.com/Which-is-grammatically-correct-which-have-or-which-has)

Reply:
Ok, will make `that have` =)